src/Facets/GenericSwapFacet.sol

```diff
@@ -7,38 +10,261 @@ import { ReentrancyGuard } from "../Helpers/ReentrancyGuard.sol";
 import { SwapperV2, LibSwap } from "../Helpers/SwapperV2.sol";
 import { Validatable } from "../Helpers/Validatable.sol";
 import { LibUtil } from "../Libraries/LibUtil.sol";
-import { InvalidReceiver } from "../Errors/GenericErrors.sol";
+import { InvalidReceiver, ContractCallNotAllowed, CumulativeSlippageTooHigh, NativeAssetTransferFailed } from "../Errors/GenericErrors.sol";
+import { IERC20, SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { LibAllowList } from "../Libraries/LibAllowList.sol";
 
-/// @title Generic Swap Facet
+/// @title GenericSwapFacet
 /// @author LI.FI (https://li.fi)
 /// @notice Provides functionality for swapping through ANY APPROVED DEX
-/// @dev Uses calldata to execute APPROVED arbitrary methods on DEXs
-/// @custom:version 1.0.0
+/// @dev Can only execute calldata for APPROVED function selectors
+/// @custom:version 2.0.0
 contract GenericSwapFacet is ILiFi, ReentrancyGuard, SwapperV2, Validatable {
+    using SafeERC20 for IERC20;
+
+    error InsufficientSwapOutput();
+
     /// External Methods ///
 
+    /// @notice Performs a single swap from an ERC20 token to another ERC20 token
+    /// @param _transactionId the transaction id associated with the operation
+    /// @param _integrator the name of the integrator
+    /// @param _referrer the address of the referrer
+    /// @param _receiver the address to receive the swapped tokens into (also excess tokens)
+    /// @param _minAmountOut the minimum amount of the final asset to receive
+    /// @param _swapData an object containing swap related data to perform swaps before bridging
+    function swapTokensSingleERC20ToERC20(
+        bytes32 _transactionId,
+        string calldata _integrator,
+        string calldata _referrer,
+        address payable _receiver,
+        uint256 _minAmountOut,
+        LibSwap.SwapData calldata _swapData // TODO: do we need refundExcessNative here?
+    ) external payable returns (uint256 amountOut) {
+        // deposit funds
+        IERC20(_swapData.sendingAssetId).safeTransferFrom(
+            msg.sender,
+            address(this),
+            _swapData.fromAmount
+        );
+
+        // ensure that contract (callTo) and function selector are whitelisted
+        if (
+            // !((LibAsset.isNativeAsset(_swapData.sendingAssetId) || // TODO: confirm that we can safely remove this
+            // LibAllowList.contractIsAllowed(_swapData.approveTo)) && // TODO: confirm that we can safely remove this
```

Comment:
I believe we can safely remove these in all cases.

Reply:
removed